src/index.ts

```diff
@@ -156,21 +160,95 @@ async function findV1Stacks(
     }
 
     if (body.Resources.CDKMetadata) {
-      let stackUsedV1 = false;
+      let stackVersion: undefined | string;
+
       if (body.Resources.CDKMetadata.Properties?.Analytics) {
         const buf = Buffer.from(body.Resources.CDKMetadata.Properties.Analytics.split(':').splice(2)[0], 'base64');
         const analyticsString = zlib.gunzipSync(buf).toString();
-        stackUsedV1 = analyticsString.slice(0, 1) === '1';
+        const constructInfo = decodePrefixEncodedString(analyticsString);
+        const stackConstruct = constructInfo.find(x => x.endsWith('@aws-cdk/core.Stack') || x.endsWith('monocdk.Stack'));
+        if (stackConstruct) {
+          stackVersion = stackConstruct.split('!')[0];
+        }
       }
 
       // Before versions 1.93.0 and 2.0.0-alpha.10, the CDKMetadata resource had a different format.
       if (body.Resources.CDKMetadata.Properties?.Modules) {
-        stackUsedV1 = body.Resources.CDKMetadata.Properties.Modules.includes('@aws-cdk/core');
+        const modules = body.Resources.CDKMetadata.Properties.Modules.split(',') as string[];
+        const coreModule = modules.find(m => m.startsWith('@aws-cdk/core=') || m.startsWith('monocdk='));
```

Comment:
How sure are we that `@aws-cdk/core@1.` cannot exist? Like it does in the metrics extractor data?

Reply:
Confirming with @corymhall that this is added by our server side processing, and wouldn't be found here.

Reply:
It's not. Looks like it's weird content from a customer who has forked and hacked the CDK library.